core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -19,122 +19,116 @@
-      this.warehouseUri = warehouseUri;
-    } else {
-      String fsRoot = conf.get("fs.defaultFS");
-      Path warehousePath = new Path(fsRoot, ICEBERG_HADOOP_WAREHOUSE_BASE);
-      try {
-        FileSystem fs = Util.getFs(warehousePath, conf);
-        if (!fs.isDirectory(warehousePath)) {
-          if (!fs.mkdirs(warehousePath)) {
-            throw new IOException("failed to create warehouse for hadoop catalog");
-          }
+  public HadoopCatalog(Configuration conf) {
+    String fsRoot = conf.get("fs.defaultFS");
+    Path warehousePath = new Path(fsRoot, ICEBERG_HADOOP_WAREHOUSE_BASE);
+
+    try {
+      FileSystem fs = Util.getFs(warehousePath, conf);
+      if (!fs.isDirectory(warehousePath)) {
+        if (!fs.mkdirs(warehousePath)) {
+          throw new IOException("failed to create warehouse for hadoop catalog");
         }
-        this.warehouseUri = fsRoot + "/" + ICEBERG_HADOOP_WAREHOUSE_BASE;
-      } catch (IOException e) {
-        throw new RuntimeIOException("failed to create directory for warehouse", e);
       }
+    } catch (IOException e) {
+      throw new RuntimeIOException("failed to create directory for warehouse", e);
     }
+
+    this.conf = conf;
+    this.warehouseLocation = fsRoot + "/" + ICEBERG_HADOOP_WAREHOUSE_BASE;
   }
 
-  public HadoopCatalog(Configuration conf) {
-    this(conf, null);
+  @Override
+  public Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, String location, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length >= 1,
+        "Missing database in table identifier: %s", identifier);
+    Preconditions.checkArgument(location == null, "Cannot set a custom location for a path-based table");
+    return createTable(identifier, schema, spec, properties);
   }
 
   @Override
-  public org.apache.iceberg.Table createTable(
+  public Table createTable(
       TableIdentifier identifier, Schema schema, PartitionSpec spec, Map<String, String> properties) {
     Preconditions.checkArgument(identifier.namespace().levels().length >= 1,
-            "Missing database in table identifier: %s", identifier);
-    Path tablePath = new Path(defaultWarehouseLocation(identifier));
-    try {
-      FileSystem fs = Util.getFs(tablePath, conf);
-      if (!fs.isDirectory(tablePath)) {
-        fs.mkdirs(tablePath);
-      } else {
-        throw new AlreadyExistsException("the directory %s for table %s is already exists.",
-          tablePath.toString(), identifier.toString());
-      }
-    } catch (IOException e) {
-      throw new RuntimeIOException("failed to create directory", e);
-    }
+        "Missing database in table identifier: %s", identifier);
     return super.createTable(identifier, schema, spec, null, properties);
   }
 
-  public org.apache.iceberg.Table createTable(
-          TableIdentifier identifier, Schema schema, PartitionSpec spec) {
+  public Table createTable(
```

Comment:
The only implementation of `createTable` that is necessary is the one with the location option.